lib/msal-node/src/client/ConfidentialClientApplication.ts

```diff
@@ -69,17 +70,23 @@ export class ConfidentialClientApplication extends ClientApplication implements
             };
         }
 
-        const validRequest: CommonClientCredentialRequest = {
+        const originalRequest: CommonClientCredentialRequest = {
             ...request,
             ...await this.initializeBaseRequest(request),
             clientAssertion
         };
 
+        // a valid client credentials request should not contain delegated permissions
+        const validRequest: CommonClientCredentialRequest = {
+            ...originalRequest,
+            scopes: originalRequest.scopes.filter(scope => !OIDC_DEFAULT_SCOPES.includes(scope))
+        };
```

Comment:
This could be done in one step if you put the result of `await this.initializeBaseRequest(request)` into a variable.

Reply:
I was also attempting to remove any oidc scopes added by the end developer, but perhaps that was unnecessary.